### ISSUE TITLE
Mismatched argument names in `.pxd` undetected.

### ISSUE THREAD

**Issue by will-ca** [cython/cython#3417]
This form silently compiles, and seems to type `kwarg` as `bint` even though it's misspelled in the `.pxd` file:
```python3
def misnamedarg(arg, kwarg):
	return (arg, kwarg)
```
```python3
cpdef tuple misnamedarg(double arg, bint quarg)
```

This form fails to compile, as it probably should, but with an opaque compiler crash (`AttributeError()`) instead of an explicit `CompileError()` and error report:
```python3
def misnamedkwarg(arg, kwarg=True):
	return (arg, kwarg)
```
```python3
cpdef (double, bint) misnamedkwarg(double arg, bint quarg=*)
```

**Comment by scoder** [comment]
Yes, this should at least produce a warning. Note that it's not required to provide argument names in the .pxd signature (as in C prototypes), but if they are there, then they should better match. (It doesn't actually matter for `cdef` functions, but for `def` and `cpdef`, it does, because of names keyword arguments.)
